spec/models/user_spec.rb

```diff
@@ -135,6 +135,53 @@
     end
   end
 
+  describe '#validations' do
+    context 'password' do
+      let(:user) { build :user }
+
+      context 'when password is blank' do
+        before do
+          user.save
+        end
+        # password is just an attribute in user, it is not a field
+        # so, password is nil, when it is not set
+        # devise will take care of nil/blank password while registering a user
+        it 'should not throw any error' do
+          user = User.first
+          expect(user.password).to be nil
+
+          expect(user.valid?).to be true
+        end
+      end
+
+      context 'when oauth is enabled' do
+        it 'saves without any errors even if the password strength is less' do
+          user.password = 'warsdasdf'
+          user.token = 'access token'
+          expect(user.valid?).to be true
+        end
+      end
+
+      context 'with valid one' do
+        it 'saves without any errors' do
+          user.password = 'waspAr$0'
+          expect(user.valid?).to be true
+        end
+      end
+
+      context 'with invalid one' do
```

Comment:
I think we can be more descriptive in this context. We are talking about passwords, so something like

```ruby
context 'when password is invalid'
....
end
```